arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Option.kt

```diff
@@ -1009,6 +1231,10 @@ public fun <A> Iterable<Option<A>>.combineAll(MA: Monoid<A>): Option<A> =
 public fun <A> Option<A>.combineAll(MA: Monoid<A>): A =
   getOrElse { MA.empty() }
 
+@Deprecated(
+  RedundantAPI + "Prefer if-else statement inside option DSL, or replace with explicit flatMap",
+  ReplaceWith("this.flatMap { b -> b.takeIf(predicate)?.let(::Some) ?: None.also(error) }")
```

Comment:
A pity we cannot do more complex rewrites... DSL based code looks so much better 😁 

```kotlin
option {
  ensureNotNull(bind().takeIf(predicate))
}.onNone(error)
```

Reply:
Do we need to add imports here?

```suggestion
  ReplaceWith("this.flatMap { b -> b.takeIf(predicate)?.let(::Some) ?: None.also(error) }", "arrow.core.Some", "arrow.core.None")
```

Reply:
Yep. I think so. Good catch!